### ISSUE TITLE
Intermittent failures in  MueLu_Maxwell3D-Tpetra-Stratimikos_MPI_4 

### ISSUE THREAD

**Issue by prwolfe** [trilinos/Trilinos#5850]
## Bug Report
@trilinos/amesos2 
@trilinos/muelu 
@trilinos/stratimikos 

### Description
This fails intermittently on CUDA builds and gcc 4.8.4 builds - see https://testing.sandia.gov/cdash/testDetails.php?test=87994601&build=5569462

Also reference #5713

### Steps to Reproduce
This is intermittent so I'm not sure it can be reliably reproduced

**Comment by prwolfe** [comment]
Ross also noted this in #3276 

From: Bartlett, Roscoe A
Sent: Monday, August 05, 2019 12:26 PM
To: trilinos-framework@software.sandia.gov
Subject: FYI: Randomly failing test MueLu_Maxwell3D-Tpetra-
Stratimikos_MPI_4

Hello Trilinos Framework team,

FYI: Looks like the test MueLu_Maxwell3D-Tpetra-Stratimikos_MPI_4 is
randomly failing in the GCC 4.8.4 build (which should match the GCC 4.8.4 PR
build). As shown at:

https://testing-
dev.sandia.gov/cdash/index.php?project=Trilinos&filtercount=3&showfilters
=1&filtercombine=and&field1=buildname&compare1=61&value1=Linux-GCC-
4.8.4-
MPI_RELEASE_DEBUG_SHARED_PT_OPENMP_CI&field2=groupname&compar
e2=61&value2=Continuous&field3=buildstarttime&compare3=83&value3=16
%20weeks%20ago

it failed twice since 7/25/2019.

This will be bringing down PR builds randomly.

Can someone create a GitHub issue to address this?

-Ross

Dr. Roscoe A. Bartlett, PhD
https://bartlettroscoe.github.io/
Sandia National Laboratories

-----Original Message-----
From: CDash admin@cdash.org
Sent: Monday, August 05, 2019 11:18 AM
To: Bartlett, Roscoe A rabartl@sandia.gov
Subject: [EXTERNAL] FAILED (t=1): Trilinos/MueLu - Linux-GCC-4.8.4-
MPI_RELEASE_DEBUG_SHARED_PT_OPENMP_CI - Continuous

A submission to CDash for the project Trilinos has failing tests.
You have been identified as one of the authors who have checked in
changes that are part of this submission or you are listed in the default
contact list.

Details on the submission can be found at
https://testing.sandia.gov/cdash/buildSummary.php?buildid=5418940

Project: Trilinos
SubProject: MueLu
Site: ceerws1113
Build Name: Linux-GCC-4.8.4-
MPI_RELEASE_DEBUG_SHARED_PT_OPENMP_CI
Build Time: 2019-08-05T14:46:26 UTC
Type: Continuous
Tests not passing: 1

Tests failing
MueLu_Maxwell3D-Tpetra-Stratimikos_MPI_4 | Completed (Failed) |
(https://testing.sandia.gov/cdash/testDetails.php?test=85726525&build=
541
8940)

-CDash on testing.sandia.gov

**Comment by ndellingwood** [comment]
@cgcgcg if this is a case where a matrix with gapped row ids is being handed to Amesos2 you can try setting the following parameter list option to handle this:
```
Teuchos::ParameterList amesos2_params("Amesos2");
amesos2_params.sublist(solverName).set("IsContiguous", false, "Are GIDs Contiguous");
```

**Comment by mhoemmen** [comment]
@ndellingwood Does Amesos2 not detect this on its own?

**Comment by ndellingwood** [comment]
@mhoemmen no, at the time we added the enhancement to handle the case for gapped rowids we did not want to require all users to pay for the communication in the iscontiguous check for what seemed to be a small number of cases where this problem occurred. 
Spoke briefly with Christian and gapped rowids is not the issue he is dealing with.

**Comment by prwolfe** [comment]
Out last failure of this that was not on rocket man was 9/4/2019 so I am tempted to say it's fixed. @cgcgcg - do you  have any idea why this is failing on rocket man? It seems to be very consistent there.

**Comment by cgcgcg** [comment]
@prwolfe Seems that all MueLu tests are failing in one of the rocketman builds. So I believe that this issue is fixed.
